02-express-tutorial/app.js

```diff
@@ -1 +1,76 @@
+const express = require("express")
+
+const app = express()
+//exercise 8
+const { products } = require("./data");
+//exercise 4
+app.use(express.static("./public"))
+//exercise 7
+app.get("/api/v1/test", (req, res)=> {
+res.status(200).json({ message: "It worked!" })
+})
+
+app.get("/api/v1/products", (req, res)=> {
+res.status(200).json( products )
+})
+
+app.get("/api/v1/products/:productID", (req, res)=> {
+    //exercise 9
+//res.status(200).json(req.params) 
+//exercise 10
+  const idToFind = parseInt(req.params.productID); // parseInt will convert a string to an integer
+  console.log(idToFind)
+const product = products.find((p) => p.id === idToFind); 
+//exercise 11
+if(!product) {
+    return res.status(404).json({ message: "That product was not found."})
+} 
+return res.status(200).json(product)
+})
+//exercise 12
+app.get("/api/v1/query", (req, res)=> {
+    console.log(req.query)
+    const {search, limit,priceLessThan} = req.query
+    let sortedItems = [...products]
+    if(search){
+        sortedItems=sortedItems.filter((product)=> {
+        return product.name.startsWith(search)
+        })
+    }
+    if(limit){
+         sortedItems=sortedItems.slice(0, Number(limit))
+    } 
+    //exercise 13
+    if(priceLessThan){
+         sortedItems=sortedItems.filter((product)=>{
+        return product.price<priceLessThan
+        })
+    }
+    if(sortedItems.length<1){
+        // res.status(200).send("no products match your search")
+       return  res.status(200).json({success: true, data: []})
+    }
+  return  res.status(200).json(sortedItems)
```

Comment:
I suggest using only one pattern of response here. When no product is found, you return an object with two fields, success and data. But when it finds products it returns only the array of products.
I liked the pattern of returning an object with fields success and data. I think you can use only one res, no matter if it finds products or not. For example: return  res.status(200).json({success: true, data: sortedItems});

Reply:
Thank you! It is a great remark. I will do it.